src/appleseed/renderer/kernel/intersection/assemblytree.cpp

```diff
@@ -173,6 +176,13 @@ void AssemblyTree::rebuild_assembly_tree()
         pretty_int(m_items.size()).c_str(),
         plural(m_items.size(), "assembly instance").c_str());
 
+    #ifdef APPLESEED_WITH_EMBREE
```

Comment:
You've put this code in the middle of the assembly tree construction, and in the middle of log messages related to that construction. Move this code somewhere else. I suggest the end of this method.